src/Test/Html/Selector.elm

```diff
@@ -208,46 +210,52 @@ For attributes with boolean values, such as `checked`, use [`boolAttribute`](#bo
             Html.div []
                 [ Html.h1 [ Attr.title "greeting" ] [ Html.text "Hello!" ] ]
                 |> Query.fromHtml
-                |> Query.find [ attribute "title" "greeting" ]
+                |> Query.find [ attribute <| Attr.title "greeting" ]
                 |> Query.has [ text "Hello!" ]
 
 -}
-attribute : String -> String -> Selector
-attribute name value =
-    Attribute
-        { name = name
-        , value = value
-        , asString = "attribute " ++ toString name ++ " " ++ toString value
-        }
-
-
-{-| Matches elements that have the given attribute with the given boolean value.
+attribute : Attribute Never -> Selector
+attribute attr =
+    let
+        name =
+            Html.Inert.attributeName attr
+
+        fakeDiv =
+            Html.div [ attr ] []
+                |> Html.Inert.fromHtml
+                |> Html.Inert.toElmHtml
```

Comment:
:thought_balloon: This seems like more work than is needed. Perhaps it would be better to expose a way to use `decodeFacts` directly